src/data/validation/email.js

```diff
@@ -155,19 +183,48 @@ const returnValidatedEmails = (formData) => {
   if (errorsDict._error.length > 0) {
     throw new SubmissionError(errorsDict);
   }
-  let emails = [];
+  let emails = extractEmails(formData).allEmails;
+  emails = _.union(emails); // Dedup emails
+  return validateEmailAddresses(emails).validEmails;
+};
+
+// Combine all the rows from textarea and CSV and then make a map of email to salesforce id
+const parseData = (formData) => {
+  const rows = [];
+  const allRecords = {};
+
   if (formData[EMAIL_ADDRESS_TEXT_FORM_DATA] && formData[EMAIL_ADDRESS_TEXT_FORM_DATA].length) {
-    emails.push(...formData[EMAIL_ADDRESS_TEXT_FORM_DATA].split(/\r\n|\n/));
+    rows.push(...formData[EMAIL_ADDRESS_TEXT_FORM_DATA].split(/\r\n|\n/));
   }
+
   if (formData[EMAIL_ADDRESS_CSV_FORM_DATA] && formData[EMAIL_ADDRESS_CSV_FORM_DATA].length) {
-    emails.push(...formData[EMAIL_ADDRESS_CSV_FORM_DATA]);
+    rows.push(...formData[EMAIL_ADDRESS_CSV_FORM_DATA]);
   }
-  emails = _.union(emails); // Dedup emails
-  return validateEmailAddresses(emails).validEmails;
+
+  rows.forEach((row) => {
+    const [email, id] = row.split(',').map(item => item.trim());
```

Comment:
nit: is it perhaps worth renaming `id` to `salesforceId` or similar, so reading the code has a bit more context added to what type of `id` it is (Salesforce).